### PR TITLE
HTS-287 : Initial HTML page for showing up missing info errors to the user

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala [hunk 1]
```diff
@@ -74,20 +75,17 @@ class RegisterController @Inject()(val messagesApi: MessagesApi,
...
-                  case _ ⇒
-                    Logger.error(s"Could not retrieve logged in user's NINO or userDetails")
-                    SeeOther(routes.RegisterController.accessDenied().url)
+                }, { success ⇒
+                  val nino = success._1
+                  val eligibility = success._2
+                  eligibility.result.fold(
+                    infos ⇒ {
+                      Logger.error(s"user $nino has missing information: ${infos.missingInfo.mkString(",")}")
+                      Ok(views.html.register.missing_user_info(infos.missingInfo, personalAccountUrl))
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala]: What details can you actually change via pertax-frontend?

File: app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala [hunk 2]
```diff
@@ -74,20 +75,15 @@ class RegisterController @Inject()(val messagesApi: MessagesApi,
...
-                      }, {
-                        case Some(info) ⇒ Ok(views.html.register.confirm_details(info))
-                        case _ ⇒ SeeOther(routes.RegisterController.notEligible().url)
-                      })
-
-                  case _ ⇒
-                    Logger.error(s"Could not retrieve logged in user's NINO or userDetails")
-                    SeeOther(routes.RegisterController.accessDenied().url)
+                }, { case (nino, eligibility) ⇒
+                  eligibility.result.fold(
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala]: agreed, wanted to get rid of ```case _ => ``` part , but anyway used case now , looks better

File: app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala (hunk 1 above)
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala]: I can see only address can be updated in the /personal-account page ...

**Comment by AndyHWChung** [comment]
Since pertax-frontend can only change the address do we need to only direct them to there if the address is missing?

**Comment by sureshhmrc** [comment]
in general its nice to provide the personal account link and ask user to ```go and update the address and then continue help-to-save account```  but either way i do not have a strong opinion on how it should look like... its upto UX guys

**Comment by AndyHWChung** [comment]
I guess the code lays down the framework code needed to direct the user to the right place, but is there somewhere where they can change the reset of the details? Probably not, but just wondering how much this has been investigated.

**Comment by AndyHWChung** [comment]
The code looks fine - I'll leave it to Marcus to comment on what the redirects should really be

**Pull request merged**